pkg/provider/identity_test.go

```diff
@@ -208,6 +211,24 @@ func TestIdentityConfig_LoadIdentity(t *testing.T) {
 	assert.Equal(t, expectedFI.Leaf, fi.Leaf)
 	assert.Equal(t, expectedFI.CA, fi.CA)
 	assert.Equal(t, expectedFI.ID.Bytes(), fi.ID.Bytes())
+
+	tmp := path.Join(os.TempDir(), "temp-ca-whitelist.pem")
+	w, err := os.Create(tmp)
+	assert.NoError(t, err)
+	defer func() {
+		err := os.RemoveAll(tmp)
+		if err != nil {
+			fmt.Printf("unable to cleanup temp ca whitelist at \"%s\": %s", tmp, err.Error())
```

Comment:
should you `assert.NoError(t, err)' here as well ?

Reply:
Can you do that in a `defer`? I don't think it's really relevant to the test though if for some reason the temp file was already cleaned up.

Reply:
yea you can -- also if you cared about it you could wait until the go func finished before completing the test. Just a nit though, no need to change.